nvflare/private/fed/server/server_command_agent.py

```diff
@@ -112,19 +112,17 @@ def aux_communicate(self, request: CellMessage) -> CellMessage:
                 make_reply(ReturnCode.AUTHENTICATION_ERROR, error, None)
 
             engine = fl_ctx.get_engine()
-            reply = engine.dispatch(topic=topic, request=data, fl_ctx=fl_ctx)
+            if not engine:
+                # I (SJ) cannot process this request because my engine is not set yet.
+                # This happens only when a CJ became ready quickly and send runner_sync request to the SJ.
+                # I'll simply tell it that my service is not available, and it will retry.
+                return make_reply(ReturnCode.SERVICE_UNAVAILABLE)
 
-            self.logger.debug("Before gen_new_peer_ctx")
-            shared_fl_ctx = gen_new_peer_ctx(fl_ctx)
-            self.logger.debug("After gen_new_peer_ctx")
-            reply.set_peer_context(shared_fl_ctx)
-
-            if reply is not None:
-                return_message = new_cell_message({}, reply)
-                return_message.set_header(MessageHeaderKey.RETURN_CODE, ReturnCode.OK)
-            else:
-                return_message = new_cell_message({}, None)
-            return return_message
+            reply = engine.dispatch(topic=topic, request=data, fl_ctx=fl_ctx)
+            if reply:
+                shared_fl_ctx = gen_new_peer_ctx(fl_ctx)
+                reply.set_peer_context(shared_fl_ctx)
+            return new_cell_message({MessageHeaderKey.RETURN_CODE: ReturnCode.OK}, reply)
```

Comment:
if reply is None, we will return a cell message with "ReturnCode.OK" is that what we want?